### PR TITLE
Cleanup: making process deposit more spec readable and fork specific

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

Currently, the `ProcessDeposit` function lives in a shared fork repo in the `blocks` package. To prepare for changes in 6110 I have broken the functions to match more closely with the consensus specs defined in https://github.com/ethereum/consensus-specs. As part of this PR I've moved the deposit functions from the blocks package to the altair package which will allow easier updating in future hard forks.

breaking out pr #13944

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

**Review by saolyn — APPROVED** [review]
LGTM, mostly just moving tests around and some renaming. Might be good to wait for someone else to have a pass before merging

File: beacon-chain/core/altair/deposit.go [hunk 1]
```diff
@@ -5,11 +5,32 @@ import (
...
 	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/helpers"
 	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/config/params"
 	"github.com/prysmaticlabs/prysm/v5/encoding/bytesutil"
 	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
 )
 
+// ProcessPreGenesisDeposits processes a deposit for the beacon state before chainstart.
```
Comment (terencechain) [inline beacon-chain/core/altair/deposit.go:16]: Why are these in `altair`? I think these additions are not part of altair hard fork right

Comment (james-prysm) [reply beacon-chain/core/altair/deposit.go:16]: the alternative would be to break out process deposits into yet another for phase 0. I considered this but just decided to have the two ( altair and electra). if you think I should still break this out I can. 
the issue with leaving it in the blocks package is the circular dependency, processPreGenesisDeposits calls process deposits which needs to specify a fork for process deposits, in this case it only makes sense to use the current version of process deposits

**Review by terencechain — APPROVED** [review]
Lgtm

**Pull request merged**